slasher/detection/filter/filter.go

```diff
@@ -0,0 +1,105 @@
+package filter
+
+import (
+	"encoding/binary"
+	"hash/fnv"
+
+	"github.com/prysmaticlabs/prysm/shared/hashutil"
+	"github.com/spaolacci/murmur3"
+)
+
+// BloomFilter is an encoded set of a []byte key.
+type BloomFilter []byte
+
+// NewBloomFilter returns a new bloom filter that encodes the given key with 16 bits allotted for it.
+// This bloom filter has a set length of 16 bits, and uses 5 hash functions in order to provide less than
+// 0.1% chance of false positives.
```

Comment:
i am afraid of false negative. i think its 0.1% for false negative . wdyt?

Reply:
I think it's considerably safe considering the chances of an real double vote qualifying for the conditions. We can revisit it later.

If needed we can also look into expanding it another uint16 which will definitely provide more safety.